### PR TITLE
Bump Pyro version to 0.3.4

### PR DESCRIPTION
**Pull request by neerajprad**

### CODE REVIEW

**Comment by fehiepsi** [comment]
Just a question: should we remove [IPython restriction](https://github.com/pyro-ppl/pyro/blob/dev/setup.py#L59) because the error is fixed with new release of ipython?

File: setup.py [hunk 1]
```diff
@@ -101,7 +99,7 @@
             'pytest-cov',
             # TODO: remove once https://github.com/pyro-ppl/pyro/issues/1871
```
Comment (fehiepsi) [inline setup.py]: nit: we can remove this comment too

**Comment by neerajprad** [comment]
This is ready to be merged. Should we wait for #1960 to be fixed first?

**Comment by fritzo** [comment]
@neerajprad ready to merge? @fehiepsi's fix for #1960 is now merged.

**Comment by neerajprad** [comment]
Yes, this is good to merge now.

**Pull request merged**